### ISSUE TITLE
[clang] Discarding a use of a [[nodiscard]] `consteval` function does not emit a nodiscard warning

### ISSUE THREAD

**Issue by smallp-o-p** [llvm/llvm-project#141536]
It emits an `expression result unused` warning instead. GCC emits the `nodiscard` warning. 

``` 
[[nodiscard]] consteval int foo(){ return 1; }

[[nodiscard]] int foo2(){ return 1; }

int main(){
    foo(); // <-- emits an expression result unused warning
    foo2(); 
}
```
See: https://godbolt.org/z/9fEG8c5Mx

**Comment by frederick-vs-ja** [comment]
I don't think the original example is buggy. Clang is more aggressive on warning on discarding results of `consteval` functions, which is arguably a good thing.

But for this example, Clang ignores the user-provided message, which is somehow undesired. [Godbolt link](https://godbolt.org/z/co8Tha447).
```C++
[[nodiscard("my message")]] consteval int foo(){ return 1; }

int main(){
    foo();
}
```
